devstack/plugin.sh

```diff
@@ -0,0 +1,148 @@
+
+# Devstack plugin code for Calico
+# ===============================
+
+function install_configure_etcd {
+
+    # Install etcd from package sources (=> our PPA).
+    install_package etcd
+
+    # Stop the etcd service:
+    sudo service etcd stop || true
+
+    # Delete any existing etcd database:
+    sudo rm -rf /var/lib/etcd/*
+
+    # Mount a RAM disk at /var/lib/etcd:
+    sudo mount -t tmpfs -o size=512m tmpfs /var/lib/etcd
+
+    # Add the following to the bottom of /etc/fstab so that the RAM
+    # disk gets reinstated at boot time:
+    # tmpfs /var/lib/etcd tmpfs nodev,nosuid,noexec,nodiratime,size=512M 0 0
+
+    IP=`hostname -I | awk '{print $1}'`
+
+    # Edit /etc/init/etcd.conf: Find the line which begins exec
+    # /usr/bin/etcd and edit it, substituting for <controller_fqdn>
+    # and <controller_ip> appropriately.
+    sudo sed -i "s/exec.*/exec \/usr\/bin\/etcd --name=\"$HOSTNAME\" \
```

Comment:
Would it be better to just use a heredoc to cat out the file rather than running a very long regex over it?